### PR TITLE
rebuild NWIS, WQP, Ecosheds data

### PR DESCRIPTION
**Pull request by limnoliver**
This PR repulls NWIS and WQP temperature data for the nation. 

**For data updates:**

https://github.com/USGS-R/2wp-temp-observations/pull/52/commits/38094d66aa80adc5f458461b8ffc86b6b1ae254b - For WQP data, note there was a gain in sites (17k) but a loss in the number of records (792k). I compared the old data file to the new data file to try to explain some of this. Note, the pull includes sub-daily data, and there was a gain in the number of unique site-days (311k). On the surface this is worrisome, but after parsing the differences, I'm not worried about 

- There were 1678 sites in the old pull that are not in the new pull. This amounts to a loss of 2.88 million records (but only 77k unique site-days).
- There were 2451 sites that had fewer records in the new pull compared to the old pull. This amounts to a loss of 321k records (but only 8k unique site-days).
- There were 15552 sites in the new pull that were not in the old pull. This amounts to a gain of 565k new records and 131k new site-dates. 
- There were 24005 sites that had a gain in the number of records from the old pull to the new pull (gain of 1.78 mil records and 265k site-dates).
- The majority of sites (341238) had no change in the number of records (but +46 site-dates).

Of the missing sites, only one of those was in in the inventory/partition files (e.g., was returned from whatWQPsites). I spot checked a few of the sites missing with the most data by looking at the WQP homepage. Some appeared to have MonitoringLocationIdentifier changes (e.g., BTMUA-INTAKE became BTMUA-Intake) while others seem to no longer exist (KENAI_WQX-10000117). It doesn't seem like we systematically missed them in our pull (e.g., a bad partition pull or whatWQPsites changes). 

https://github.com/USGS-R/2wp-temp-observations/pull/52/commits/b0017c9926766f64e866311bc1f6266536eb0453 - expected increases in sites and data from NWIS.

**For moderate code changes:**

https://github.com/USGS-R/2wp-temp-observations/pull/52/commits/42086abf3ce564d562fab641d12f43b23db14a00 - retaining all temperature columns that are returned from NWIS instead of a priori selecting one.

https://github.com/USGS-R/2wp-temp-observations/pull/52/commits/909988d4f021d798a916410b26bfd783a2d70daf - pulls HUC22 which includes US territories, https://github.com/USGS-R/2wp-temp-observations/pull/52/commits/ef1784d99cce07419b3d9ebe93bff681bb4bf2be modifies some spatial filters to ensure those site are retained.

https://github.com/USGS-R/2wp-temp-observations/pull/52/commits/682466f209cd86999228f352d6a63e8e9750a270 - use .qs files instead of .rds files for temporary files. Forgot to do this with the WQP pull -- will do this [next time](https://github.com/USGS-R/2wp-temp-observations/issues/57)!

https://github.com/USGS-R/2wp-temp-observations/pull/52/commits/9bd0aa80e8d6a3389a7ce7d54caf99decb7e4d3b - modified the task tables for WQP pulls to include the pull date so new pulls would be triggered.

### CODE REVIEW

**Comment by limnoliver** [comment]
This now includes an Ecosheds rebuild after receiving a new snapshot of Ecosheds from Jeff Walker. This [produced increases](https://github.com/USGS-R/2wp-temp-observations/blob/af5958393b2142b55fca299171263e2dbdd8f019/4_other_sources/out/ecosheds_summary.csv) in both records and sites as expected.

**Comment by padilla410** [comment]
Looks good to me, Sam. Only one question on retaining some commented-out code. 

The use of `qs` and the updates required to get the territories are similar to the work on the `national-flow-repo` (minus the spatial work).

Side note, I converted this PR from draft to review.

**Review by padilla410 — APPROVED** [review]

**Comment by padilla410** [comment]
Looks good. Without running it, it looks like it does what you're saying.

File: 2_nwis_pull/src/nwis_pull.R [hunk 1]
```diff
@@ -174,31 +174,31 @@ choose_temp_column_dv <- function(temp_dat) {
...
-    group_by(site_no, location_info) %>%
-    mutate(count_nu = n()) %>%
-    ungroup() %>%
-    group_by(site_no, Date) %>%
-    slice(which.max(count_nu)) %>%
-    ungroup() %>%
-    select(-agency_cd, -count_nu)
-
-  if (!all(names(fixed_dups) %in%
+  # fixed_dups <- dat_long %>%
```
[the remark this answers is not in the document]
Comment (limnoliver) [reply 2_nwis_pull/src/nwis_pull.R:185]: Thanks for spotting this. I think I just did this during development, but will delete now.

[the remark this answers is not in the document]
Comment (limnoliver) [reply 2_nwis_pull/src/nwis_pull.R:185]: Actually, keeping in commented code so targets stay "clean".

**Pull request merged**